src/server.rs

```diff
@@ -577,3 +589,18 @@ impl From<models::ValidationError> for Error {
         Self::Validation(value.to_string())
     }
 }
+
+async fn filter_header(config: &OrchestratorConfig, headers: HeaderMap) -> HeaderMap {
+    let allowed_headers = config
+        .allowed_headers_passthrough
+        .clone()
+        .unwrap_or_default();
+
+    headers
+        .iter()
+        .filter(|(header_name, _)| allowed_headers.contains(&header_name.to_string()))
+        .fold(HeaderMap::new(), |mut acc, (header_name, header_val)| {
+            acc.append(header_name, header_val.clone());
+            acc
+        })
+}
```

Comment:
Unnecessary async and excess allocations. Suggested change factoring in changes above. 

NOTE: can potentially be further improved to avoid clones entirely (need to review `HeaderMap` impl), but this should be ok.

```suggestion
/// Filter request headers to configured passthrough headers.
fn filter_headers(passthrough_headers: &HashSet<String>, headers: HeaderMap) -> HeaderMap {
    headers
        .iter()
        .filter(|(name, _)| passthrough_headers.contains(name.as_str()))
        .map(|(name, value)| (name.clone(), value.clone()))
        .collect()
}
```